### DISCUSSION TITLE
8.14.2 -> 8.14.3 breaks testing on Github Actions w/ default MySQL service

### DISCUSSION THREAD

**Discussion by swichers** [orchestral/testbench-core#148] [General]
We run an automated process on multiple repos. Recently they started breaking after updating testbench. 

They all fail with this error:

```
Illuminate\Database\QueryException: SQLSTATE[HY000]: General error: 1273 Unknown collation: '' (Connection: mysql, SQL: create database `laravel_test_1` default character set `utf8mb4` default collate ``)
```

I tracked this down to a change in this repo where the database config was modified.

https://github.com/orchestral/testbench-core/commit/bc7ff4f5e77d7c6849a17b29a6f86e4335955bdb#diff-baff2ba45651d125698c0061bd2e436cc6cff8efb60485cabe48f3ff27b7ebe1R56

**Comment by crynobone** [comment]
The configuration changes inline with skeleton changes made to `laravel/laravel`.

**Reply by crynobone** [reply]
https://github.com/laravel/laravel/pull/6241

**Reply by crynobone** [reply]
Also do note that Laravel Framework also runs the latest 8.14.3 for integration tests and tested with both MySQL 5.7 and 8.

**Comment by swichers** [comment]
Very barebones project showing the failure

https://github.com/swichers/testbench-core-error

8.14.2 -> success https://github.com/swichers/testbench-core-error/actions/runs/6728476030

8.14.3 -> fail https://github.com/swichers/testbench-core-error/actions/runs/6728486529

Only difference between the two https://github.com/swichers/testbench-core-error/pull/1/commits/16c97789046d82f56308316f675fad65ca894c3f#diff-f37acfaa6b11f575a9a6f41a75fa73a61d0f8ebc2c9b8cddc215d8aca10e44f5L2969

**Reply by crynobone** [reply]
You need to take this issue to https://github.com/laravel/laravel/pull/6241 

Testbench Core only maintain skeleton configuration based on `laravel/laravel`

**Comment by swichers** [comment]
Clean fork of Laravel 10 branch passes with the same bare minimum setup

PR of changes https://github.com/swichers/laravel-test-mysql/pull/1

Test result https://github.com/swichers/laravel-test-mysql/actions/runs/6728729293/job/18288486022

If this was an issue with Laravel core shouldn't this same test setup fail?

**Reply by crynobone** [reply]
Your testbench error using `--parallel` which needs to create sub-database (which causes this error). `--parallel` ability is provided by Collision and Paratest.

**Reply by swichers** [reply]
:+1: Can confirm removing `--parallel` allows the test to run successfully.

**Comment by crynobone** [comment]
See https://github.com/laravel/framework/pull/48886

**Comment by swichers** [comment]
Removing `--parallel` as a workaround will be fine until your PR is in to Laravel core. Thanks for resolving the issue.
